pkg/metrics/metrics.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package metrics
+
+// This package defines site metrics measures and views based on opencensus.io
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+)
+
+func init() {
+	// call the actual metric provider functions for the latest metrics every 4th second
+	go func() {
+		rand.Seed(time.Now().UnixNano())
+		for {
+			getNumUsers()
+			getNumGroups()
+			getAmountStorage()
+			time.Sleep(4 * time.Second)
+		}
+	}()
+}
+
+// Create the measures
+var (
+	NumUsersMeasure      = stats.Int64("cs3_org_sciencemesh_site_total_num_users", "The total number of users within this site", stats.UnitDimensionless)
+	NumGroupsMeasure     = stats.Int64("cs3_org_sciencemesh_site_total_num_groups", "The total number of groups within this site", stats.UnitDimensionless)
+	AmountStorageMeasure = stats.Int64("cs3_org_sciencemesh_site_total_amount_storage", "The total amount of storage used within this site", stats.UnitDimensionless)
```

Comment:
Use stats.UnitBytes here

Reply:
Maybe only on storage amount ?

Reply:
Yeah, just on storage. Github shows the whole blob even if you comment on a single line.